src/ducks/recurrence/service.js

```diff
@@ -129,15 +131,23 @@ const main = async ({ client }) => {
     )
     const transactions = await client.queryAll(transactionQuery)
 
+    // Filter out transactions already associated with a recurrence as we don't
+    // want to change it.
+    const newTransactions = transactions.filter(
+      transaction =>
+        transaction.relationships == null ||
+        transaction.relationships.recurrence == null
+    )
```

Comment:
Why not using a mango selector or a partial index for that?

Reply:
We discussed this point with @nono and figured we might be better off filtering the transactions in the service since we have good bandwidth and a low latency on workers.
Do you see an efficient way to build a mango query for this? It looks like we'd need to use an `$or` operator which is quite slow if my memory serves me well.

Reply:
> We discussed this point with @nono and figured we might be better off filtering the transactions in the service since we have good bandwidth and a low latency on workers.

It depends if there is a lot of transactions with a recurrence assigned. There is a lot of instances with 6/10 years of bankings transactions. Fetching 1000 per 1000 is not efficient at all : 
- for the total duration of fetching all the transactions 
- for the total duration of the recurrence service
- for the RAM that takes on the worker 

The first 2 are a concern to me since the recurrence service is made to be call pretty often and it takes a lot of time to execute. If we can do better, we should try even if this is to win only an execution round

But yes, `$or` will make a full scan in memory... I'll not block this PR since I don't have any time to think about it, maybe @paultranvan will have an idea, but we can merge it and improve later if we can

Reply:
Old instances shouldn't be an issue here. The query only fetches transactions from the last 100 days. There shouldn't be many instances with more than 10 transactions per day on average so we should rarely do more than one request.

Reply:
Why not a partial filter ? 
```js
Q('io.cozy.banks.transactions')
  .where({
      date: {
        $gt: 100DaysBefore
      }
  })
  .partialFilter({
      'relationships.recurrence': {
        $exists: false
    })
  ```
  
With a partial filter, there is no performance issue at querying time since the filter is applied at indexing time. Also, I assume you only need to evaluate if there is no recurrence relationship? If yes, you just need the `$exists` predicate on `relationships.recurrence`.

Reply:
Thanks for the tip. This is done in https://github.com/cozy/cozy-banks/pull/2610.